drivers/dma/dma_sedi.c

```diff
@@ -0,0 +1,446 @@
+#ifdef IMR_TO_MEMORY
+	case IMR_TO_MEMORY:
+		*src_mem = DMA_UMA_MEM;
+		break;
+#endif
+	}
+}
+
+/* config basic dma */
+static int dma_sedi_apply_common_config(sedi_dma_t dev, uint32_t channel,
+					struct dma_config *config, uint8_t *dir)
+{
+	uint8_t direction = MEMORY_TO_MEMORY;
+	dma_memory_type_t src_mem = DMA_SRAM_MEM, dst_mem = DMA_SRAM_MEM;
+
+	dma_config_convert(config, &src_mem, &dst_mem, &direction);
+
+	if (dir) {
+		*dir = direction;
+	}
+
+	/* configure dma transferring direction*/
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_DIRECTION,
+			 direction);
+
+	if (direction == MEMORY_TO_MEMORY) {
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_SR_MEM_TYPE,
+				 src_mem);
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_DT_MEM_TYPE,
+				 dst_mem);
+	} else if (direction == MEMORY_TO_PERIPHERAL) {
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_HS_DEVICE_ID,
+				 config->dma_slot);
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_HS_POLARITY,
+				 DMA_HS_POLARITY_HIGH);
+		sedi_dma_control(dev, channel,
+				 SEDI_CONFIG_DMA_HS_DEVICE_ID_PER_DIR,
+				 DMA_HS_PER_TX);
+	} else if (direction == PERIPHERAL_TO_MEMORY) {
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_HS_DEVICE_ID,
+				 config->dma_slot);
+		sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_HS_POLARITY,
+				 DMA_HS_POLARITY_HIGH);
+		sedi_dma_control(dev, channel,
+				 SEDI_CONFIG_DMA_HS_DEVICE_ID_PER_DIR,
+				 DMA_HS_PER_RX);
+	} else {
+		return -1;
+	}
+	return 0;
+}
+
+static int dma_sedi_apply_single_config(sedi_dma_t dev, uint32_t channel,
+					struct dma_config *config)
+{
+	int ret = 0;
+	uint32_t temp = 0;
+
+	ret = dma_sedi_apply_common_config(dev, channel, config, NULL);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	/* configurate dma width of source data*/
+	ret = width_index(config->source_data_size, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_SR_TRANS_WIDTH, temp);
+
+	/* configurate dma width of destination data*/
+	ret = width_index(config->dest_data_size, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_DT_TRANS_WIDTH, temp);
+
+	/* configurate dma burst size*/
+	ret = burst_index(config->source_burst_length, &temp);
+	if (ret != 0) {
+		goto INVALID_ARGS;
+	}
+	sedi_dma_control(dev, channel, SEDI_CONFIG_DMA_BURST_LENGTH, temp);
+	return 0;
+
+INVALID_ARGS:
+	return ret;
+}
+
+static int dma_sedi_chan_config(const struct device *dev, uint32_t channel,
+				struct dma_config *config)
+{
+	if ((dev == NULL) || (channel >= DEV_CFG(dev)->chn_num)
+		|| (config == NULL)
+		|| (config->block_count != 1)) {
+		goto INVALID_ARGS;
+	}
+
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	struct dma_config *local_config = &(data->dma_configs[channel]);
+
+	memcpy(local_config, config, sizeof(struct dma_config));
+
+	local_config->head_block = config->head_block;
```

Comment:
no need, just remove, copy already on line 259.

Reply:
remove it.